File: Exercise3/CSharp/Exercise3_Solution/Exercise3_Solution/AttributesForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

using OSIsoft.AF.Asset;

namespace Exercise3_Solution
{
    public partial class AttributesForm : Form
    {
        private AFElement Element { get; }

        public AttributesForm(AFElement element)
        {
            Element = element;
            InitializeComponent();

            if (element != null)
            {
                lblElement.Text = element.GetPath(element.Database);
                afViewControl1.AFSetObject(element, null, null, null);
                // This should jump view to "Attributes" tab.
                afViewControl1.AFSelection = element.Attributes;
            }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: Exercise2/VB.NET/Exercise2_Solution/Exercise2_Solution/AttributesForm.vb
Imports OSIsoft.AF.Asset

Public Class AttributesForm
    Inherits Form

    Public Sub New(element As AFElement)
        InitializeComponent()

        If (element IsNot Nothing) Then
            lblElement.Text += " " + element.GetPath(element.Database)
            AfViewControl1.AFSetObject(element, Nothing, Nothing, Nothing)
            ' This should jump over to "Attributes" tab
            AfViewControl1.AFSelection = element.Attributes
        End If
    End Sub

End Class